src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs

```diff
@@ -38,6 +39,45 @@ public class MasterTestRunner : AbstractTestRunner, ITestRunner
 
         public bool IsTestRunning { get; private set; }
 
+        /// <summary>
+        /// Gets a dictionary of the package settings with defaults resolved by the engine.
+        /// </summary>
+        public IDictionary<string, object> EffectiveSettings
+        {
+            get 
+            {
+                if (_effectiveSettings == null)
+                {
+                    _effectiveSettings = new Dictionary<string, object>();
+                    foreach (string key in TestPackage.Settings.Keys)
+                        _effectiveSettings.Add(key, TestPackage.Settings[key]);
+
+                    bool processModelSpecified = _effectiveSettings.ContainsKey(PackageSettings.ProcessModel);
+                    string processModel = processModelSpecified
+                        ? (string)_effectiveSettings[PackageSettings.ProcessModel]
+                        : TestPackage.SubPackages.Count > 1
+                            ? "Multiple"
```

Comment:
I think the standard should probably say "don't use it unless it makes the code easier to read."

In this case, it did - you should see how involved it got before I did this!